### PR TITLE
[Keyboard] Capsunlocked CU80 - added variant's for RGB matrix support

### PR DESCRIPTION
**Pull request by maz0r**
## Description

*Not sure if this is the right way to add rgb matrix as an option to an existing board*

Intention is to add RGB matrix support as an option for users who wish to use per key effects, as the v2_ansi and v2_iso is setup for underglow effects.

Duplicated existing structure for v2_ansi / v2_iso

capsunlocked/cu80/v2_ansi_rgb 

- config.h
  - Updated PRODUCT_ID & PRODUCT
  - #define RGB_MATRIX_KEYPRESSES
- readme.me
  - added comment about RGB matrix support
- rules.mk
  - set RGBLIGHT_ENABLE = NO
  - RGB_MATRIX_ENABLE = yes
  - RGB_MATRIX_DRIVER = WS2812 
- v2_ansi_rgb.c
  - added code to scan matrix
  - added caps lock and scroll lock indicator code
  - added rgb matrix mapping/layout

capsunlocked/cu80/v2_iso_rgb 

- config.h
  - Updated PRODUCT_ID & PRODUCT
  - #define RGB_MATRIX_KEYPRESSES
- readme.me
  - added comment about RGB matrix support
- rules.mk
  - set RGBLIGHT_ENABLE = NO
  - RGB_MATRIX_ENABLE = yes
  - RGB_MATRIX_DRIVER = WS2812 
- v2_iso_rgb.c
  - added code to scan matrix
  - added caps lock and scroll lock indicator code
  - added rgb matrix mapping/layout

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/capsunlocked/cu80/v2_ansi_rgb/v2_ansi_rgb.c
Comment (drashna) [inline keyboards/capsunlocked/cu80/v2_ansi_rgb/v2_ansi_rgb.c:29]: Isn't used, can be removed. 
```suggestion
```

File: keyboards/capsunlocked/cu80/v2_ansi_rgb/v2_ansi_rgb.c
Comment (drashna) [inline keyboards/capsunlocked/cu80/v2_ansi_rgb/v2_ansi_rgb.c:36]:
```suggestion
```

File: keyboards/capsunlocked/cu80/v2_ansi_rgb/v2_ansi_rgb.c
Comment (drashna) [inline keyboards/capsunlocked/cu80/v2_ansi_rgb/v2_ansi_rgb.c:45]:
```suggestion
    if (host_keyboard_led_state().caps_lock) {
        rgb_matrix_set_color(62, 255, 255, 255);
        }
    if (host_keyboard_led_state().scroll_lock) {
        rgb_matrix_set_color(14, 255, 255, 255);
    }
```

File: keyboards/capsunlocked/cu80/v2_iso_rgb/v2_iso_rgb.c
Comment (drashna) [inline keyboards/capsunlocked/cu80/v2_iso_rgb/v2_iso_rgb.c:35]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by maz0r** [comment]
I'm not sure why travis started failing, but it's failing following a revert, to previously working code?

File: keyboards/capsunlocked/cu80/v2_ansi_rgb/rules.mk
Comment (fauxpark) [inline keyboards/capsunlocked/cu80/v2_ansi_rgb/rules.mk:24]:
```suggestion
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
RGB_MATRIX_ENABLE = yes     # Enable RGB matrix
```

File: keyboards/capsunlocked/cu80/v2_iso_rgb/rules.mk
Comment (fauxpark) [inline keyboards/capsunlocked/cu80/v2_iso_rgb/rules.mk:24]:
```suggestion
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
RGB_MATRIX_ENABLE = yes     # Enable RGB matrix
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/capsunlocked/cu80/v2_iso_rgb/rules.mk
Comment (fauxpark) [inline keyboards/capsunlocked/cu80/v2_iso_rgb/rules.mk:22]:
```suggestion
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/capsunlocked/cu80/v2_ansi_rgb/rules.mk
Comment (fauxpark) [inline keyboards/capsunlocked/cu80/v2_ansi_rgb/rules.mk:22]:
```suggestion
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thank you for your Contribution!

**Comment by sigprof** [comment]
Sorry, I missed this PR and saw it only after it was merged, and now I see some potential problems with it, which you may want to fix with a separate PR.

The brightness limiting that was done with `#define RGBLIGHT_LIMIT_VAL 50` is not handled by the RGB Matrix code — the proper option for RGB Matrix would be `RGB_MATRIX_MAXIMUM_BRIGHTNESS`. This may be important to avoid power issues.

(Note that `rgb_matrix_set_color()` bypasses the limit set by `RGB_MATRIX_MAXIMUM_BRIGHTNESS`, but that would probably be OK for the usage here, because only 2 indicator LEDs would be at the full brightness.)

LED flags are not 100% correct — 1 (`LED_FLAG_MODIFIER`) should be set for LEDs under the modifier keys like Ctrl, Shift, Alt (some effects use that flag to color the modifier keys differently from the normal ones). Setting this flag for Caps Lock and Scroll Lock looks wrong, and technically it is not even needed, because `rgb_matrix_indicators_kb()` will override the color of those LEDs irrespective of the flags used for them (except that you don't override their colors when the corresponding indicators are off, so in that case the current RGB Matrix effect would show there).

(Also I wonder whether the `rgb_matrix_indicators_kb()` API is really good — the issue is that there is no way to override that function from the keymap, so if someone wants those LEDs to display the RGB Matrix effect instead of working as indicators, there would be no clean way to disable the indicator code apart from patching the keyboard-level code. But that looks more like a higher-level problem which is not really specific to this particular keyboard. You could move the indicator code to the keymap level, but that would leave the configurator users without those indicators, and you might need to duplicate that code in the `via` keymap too. Another way to hack around this problem is a custom `#ifdef` around `rgb_matrix_indicators_kb()`, but it does not look like a proper general solution.)

In addition, I think that a better way to add the RGB Matrix support would be by modifying the existing `capsunlocked/cu80/v2_iso` and `capsunlocked/cu80/v2_ansi` keyboards by adding the RGB Matrix code there inside `#ifdef RGB_MATRIX_ENABLE`, and then adding `capsunlocked/cu80/v2_iso/rgb` and `capsunlocked/cu80/v2_ansi/rgb` models with just some changed `rules.mk` options and no real code:
```makefile
RGBLIGHT_ENABLE = no
RGB_MATRIX_ENABLE = yes
```
Doing it this way would avoid the code duplication (even the keymaps could be reused). But this might be too late to fix now, and other maintainers may have different opinions.

BTW, speaking of keymaps, having `RGB_SPI` and `RGB_SPD` keycodes somewhere would be useful for RGB Matrix (in a pinch you can get by with just `RGB_SPI`, because holding Shift will reverse the direction of parameter change for RGB keycodes).

**Pull request merged**
